include/bluetooth/hci.h

```diff
@@ -1914,6 +1915,28 @@ int bt_hci_cmd_send(u16_t opcode, struct net_buf *buf);
 int bt_hci_cmd_send_sync(u16_t opcode, struct net_buf *buf,
 			 struct net_buf **rsp);
 
+/** @typedef bt_hci_vnd_evt_cb_t
+  * @brief Callback type for vendor handling of HCI Vendor-Specific Events.
+  *
+  * A function of this type is registered with bt_hci_register_vnd_evt_cb()
+  * and will be called for any HCI Vendor-Specific Event. If the function
+  * can decode and handle the event it must return 0, otherwise -EINVAL.
+  *
+  * @param buf Buffer containing event parameters.
+  *
+  * @return 0 on success or -EINVAL value on failure to handle.
```

Comment:
@thoh-ot @jhedberg 
What if one wants to combine Zephyr or Mesh VS events with additional manufacturer-specific events? I would add another return value `-ENOTSUP` that then is described as "defer the handling of this vendor-specific event to Zephyr's Bluetooth stack", so that an application that wants to combine all of them can do that in a documented way, and parse only the ones that Zephyr doesn't know about, delegating the others to the stack

Reply:
> @thoh-ot @jhedberg
> What if one wants to combine Zephyr or Mesh VS events with additional manufacturer-specific events? I would add another return value `-ENOTSUP` that then is described as "defer the handling of this vendor-specific event to Zephyr's Bluetooth stack", so that an application that wants to combine all of them can do that in a documented way, and parse only the ones that Zephyr doesn't know about, delegating the others to the stack

That is what the -EINVAL error code is supposed to mean. If the callback returns -EINVAL the stack should then look at the HCI event and see if it can decode/handle it.

I can replace -EINVAL with -ENOTSUP if that is what you want.

Reply:
Right but what about the case where it's a manufacturer event, not known to the Zephyr BLE stack, but could not be processed correctly by the hook? Maybe it was malformed, maybe it had invalid data. That's why I was proposing to have an exact check on `ENOTSUP`